File: AccuPay/Core Forms/TimeEntry/Shift.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("shift")>
    Public Class Shift

        <Key>
        Public Property RowID As Integer?

        Public Property OrganizationID As Integer?

        Public Property Created As Date

        Public Property CreatedBy As Integer?

        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        Public Property TimeFrom As TimeSpan

        Public Property TimeTo As TimeSpan

        Public Property BreaktimeFrom As TimeSpan?

        Public Property BreaktimeTo As TimeSpan?

        Public Property DivisorToDailyRate As Decimal

        Public ReadOnly Property HasBreaktime As Boolean
            Get
                Return BreaktimeFrom.HasValue And BreaktimeTo.HasValue
            End Get
        End Property

        Public Sub New()
        End Sub

        Public Sub New(timeFrom As TimeSpan, timeTo As TimeSpan)
            Me.TimeFrom = timeFrom
            Me.TimeTo = timeTo
        End Sub

        Public Sub New(timeFrom As TimeSpan, timeTo As TimeSpan, breaktimeFrom As TimeSpan, breaktimeTo As TimeSpan)
            Me.New(timeFrom, timeTo)
            Me.BreaktimeFrom = breaktimeFrom
            Me.BreaktimeTo = breaktimeTo
        End Sub

    End Class

End Namespace

File: Accupay.Data/Entities/Shift.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Data.Entities
{
    [Table("shift")]
    public class Shift
    {
        [Key]
        public int? RowID { get; set; }

        public int? OrganizationID { get; set; }

        public DateTime Created { get; set; }

        public int? CreatedBy { get; set; }

        public DateTime? LastUpd { get; set; }

        public int? LastUpdBy { get; set; }

        public TimeSpan TimeFrom { get; set; }

        public TimeSpan TimeTo { get; set; }

        public TimeSpan? BreaktimeFrom { get; set; }

        public TimeSpan? BreaktimeTo { get; set; }

        public decimal DivisorToDailyRate { get; set; }

        public decimal ShiftHours { get; set; }

        public decimal WorkHours { get; set; }

        public bool HasBreaktime => BreaktimeFrom.HasValue && BreaktimeTo.HasValue;

        public Shift()
        {
        }

        public Shift(TimeSpan timeFrom, TimeSpan timeTo)
        {
            this.TimeFrom = timeFrom;
            this.TimeTo = timeTo;
        }

        public Shift(TimeSpan timeFrom, TimeSpan timeTo, TimeSpan breaktimeFrom, TimeSpan breaktimeTo) : this(timeFrom, timeTo)
        {
            this.BreaktimeFrom = breaktimeFrom;
            this.BreaktimeTo = breaktimeTo;
        }
    }
}